src/Upgrade.php

```diff
@@ -0,0 +1,54 @@
+<?php namespace Tancredi\Upgrades;
+
+require_once '../vendor/autoload.php';
+use Pimple\Container;
+
+$container = new Container();
+$container['config'] = $config;
+$container['logger'] = function($c) {
+    return \Tancredi\LoggerFactory::createLogger('Upgrader', $c);
+};
+
+$container['storage'] = function($c) {
+    global $config;
+    $storage = new \Tancredi\Entity\FileStorage($c['logger'],$config);
+    return $storage;
+};
+
+$container['UpgradeHelper'] = function($c) {
+    $helper = new \Tancredi\Entity\UpgradeHelper($c['logger']);
+    return $helper;
+};
+
+# 
+$target_version = getVersion($current_version = false);
+
+while ($target_version > getVersion()) {
+    $current_version = getVersion();
+    $next_version = $current_version + 1;
+    $container['logger']->info("Upgrading from $current_version to $next_version...");
+
+    # Launch update scripts
+    if (class_exists("\\Tancredi\\Upgrades\\Upgrader" . $next_version)) {
+        $upgrader = "\\Tancredi\\Upgrades\\Upgrader" . $next_version;
+        $upgraderObj = new $upgrader($container['storage'], $container['config'], $container['logger']);
+        $results = $upgraderObj();
+    }
+
+    # Increment current version
+    break;
+    $scope = new \Tancredi\Entity\Scope('defaults', $container['storage'], $container['logger']);
+    $scope->metadata['version'] = $next_version ;
+    $scope->setVariables();
+}
+
+
+function getVersion($current_version = true) {
```

Comment:
The argument of this function is never passed. What is it for?

Reply:
It is passed here https://github.com/nethesis/tancredi/blob/8dabbd2c5f6d4ab9eefe7ca78579696cb8938eca/src/Upgrade.php#L24 to get the starting version (the one in read write directory)
Maybe code isn't clear here, I'm open for suggestions